tdiff_path/array_feedline_paths.py

```diff
@@ -82,236 +82,256 @@
+        phase = dp['phase_deg']
+        time_ns = phase * 10**9 / (freq * 360.0)
+        dp['time_ns'] = time_ns
+
 
+    ######################################################################################
+    # Writing the array difference and combined arrays to file
     if time_file_str != 'None':
-        array_diff.tofile(plot_location + time_file_str, sep="\n")
+        array_diff.tofile(plot_location + path_type + time_file_str, sep="\n")
     if time_file_loc != 'None':
         for ant, array in all_data.items():
-            array.tofile(plot_location + time_file_loc + ant + '.txt', sep="\n")
-        combined_main_array.tofile(plot_location + time_file_loc + 'main_array_combined.txt', sep="\n")
-        combined_intf_array.tofile(plot_location + time_file_loc + 'intf_array_combined.txt', sep="\n")
-    # PLOTTING
+            array.tofile(plot_location + time_file_loc + path_type + ant + '.txt', sep="\n")
+        unwrapped_main_array.tofile(plot_location + time_file_loc + path_type +
+                            'main_array_combined.txt', sep="\n")
+        unwrapped_intf_array.tofile(plot_location + time_file_loc + path_type +
+                            'intf_array_combined.txt', sep="\n")
 
-    numplots = 6
-    fig, smpplot = plt.subplots(numplots, sharex=True, figsize=(18, 24))
+    ######################################################################################
+    # PLOTTING
+    numplots = 7
+    plot_num = 0
+    fig, smpplot = plt.subplots(numplots, 1, sharex='all', figsize=(18, 24),
+                                gridspec_kw={'height_ratios': [2, 2, 1, 1, 1, 1, 1]})
     xmin, xmax, ymin, ymax = smpplot[0].axis(xmin=8e6, xmax=20e6)
-    smpplot[numplots - 1].set_xlabel('Frequency (Hz)')
-    smpplot[0].set_title(plot_title)
-    smpplot[0].plot(combined_main_array['freq'], combined_main_array['phase_deg'] % 360.0,
-                    color=hex_dictionary['M0'], label='Main Array')
-    smpplot[1].plot(combined_main_array['freq'], combined_main_array['magnitude'],
-                    color=hex_dictionary['M0'], label='Main Array')
-    smpplot[0].plot(combined_intf_array['freq'], combined_intf_array['phase_deg'] % 360.0,
-                    color=hex_dictionary['I0'], label='Intf Array')
-    smpplot[1].plot(combined_intf_array['freq'], combined_intf_array['magnitude'],
-                    color=hex_dictionary['I0'], label='Intf Array')
-
-    smpplot[0].set_ylabel('S12 Phase of\nArrays [degrees]')  # from antenna to feedline end at building.
-    smpplot[1].set_ylabel('Combined\nArray [dB]')  # referenced to power at a single antenna
+    smpplot[numplots - 1].set_xlabel('Frequency (Hz)', size=25.0)
+    print("plotting")
 
-    for plot in range(0, numplots):
-        smpplot[plot].grid()
-    print "plotting"
-    smpplot[2].plot(array_diff['freq'], array_diff['phase_deg'])
-    smpplot[2].set_ylabel('S12 Phase\nDifference Between\nArrays [degrees]')
 
+    # PLOT: combined arrays dB and phase.
```

Comment:
I'm gonna skim plotting, but you could maybe add a comment before each section briefly describing what you are adding to the plot so its easy to track down if you want to add/change it later.